.github/workflows/hardware-long.yml

```diff
@@ -7,7 +7,7 @@ on:
   workflow_dispatch:
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}${{ github.job }}
```

Comment:
I don't think we want concurrency groups with the job ID for these jobs- if we add this, than a force push over a pull request won't cancel the previous job, will it? If you tested that and the job cancellation works, than I'm fine with this change. We do need it for the build-fw job though